library/tempfile/create_spec.rb

```diff
@@ -0,0 +1,114 @@
+require_relative '../../spec_helper'
+require 'tempfile'
+
+describe "Tempfile.create" do
+  after :each do
+    if @tempfile
+      @tempfile.close
+      File.unlink(@tempfile.path) if File.file?(@tempfile.path)
+    end
+  end
+
+  it "returns a new, open File instance placed in tmpdir" do
+    @tempfile = Tempfile.create
+    # Unlike Tempfile.open this returns a true File,
+    # but `.should be_an_instance_of(File)` would be true either way.
+    @tempfile.instance_of?(File).should be_true
+
+    @tempfile.should_not.closed?
+    File.file?(@tempfile.path).should be_true
+
+    @tempfile.path.should.start_with?(Dir.tmpdir)
+    @tempfile.path.should_not == "#{Dir.tmpdir}/"
+  end
+
+  it "returns a writable, readable file" do
+    @tempfile = Tempfile.create
+    File.writable?(@tempfile.path).should be_true
+    @tempfile << "Test!\nMore test!"
+    @tempfile.seek(0)
+    @tempfile.gets.should == "Test!\n"
+  end
```

Comment:
It seems there is a mixing of a bit different file properties:
- file permissions, that are documented as `Permissions are 0600` and partially checked with `File.writable?` and can be checked also with `f.stat.mode.to_s(8)     # => "100600"` (taken from the documentation)
- and a file open mode - `w+` what is checked with `#<<` and `#gets`

So I would divide this case into separate ones.

> Mode is 'w+' (read/write mode, positioned at the end).

Also it makes sense to check that the current position is in the end of file